src/Components/Directory.Build.targets

```diff
@@ -27,6 +27,12 @@
     </ItemGroup>
   </Target>
 
+  <Target Name="ValidatePreReleaseLabelIsCorrectForPreviewPackages"
+          AfterTargets="Pack"
+          Condition="'$(OfficialBuildId)' != '' and '$(SuppressFinalPackageVersion)' == 'true' and !$(PreReleaseVersionLabel.StartsWith('preview'))">
```

Comment:
Would it be better to actually look at the outputted nupkg and verify its version?

Reply:
This would be possible I think, yes, via a Regex that extracts the label. Unless you feel strongly about it, I can make those changes in main along with your refactoring suggestion in favor of getting rc1 build ready.